include/avif/internal.h

```diff
@@ -786,6 +786,37 @@ avifResult avifFindMinMaxWithoutOutliers(const float * gainMapF, int numPixels,
 
 #endif // AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP
 
+// ---------------------------------------------------------------------------
+// Internal encode
```

Comment:
(This is a comment on the pull request.) I probably would add the generated test images to the source tree rather than generate them on the fly during testing.

This PR consists of two parts. The `avifEncoderInternalOptions` part is straightforward. On the other hand, it is also easy to rewrite this part from scratch when necessary -- one just needs to make simple changes to the `avifWriteToneMappedImagePayload()` function. So it is less valuable to check in this part.

The `avifEncoderAddImageInternal` part is more complicated. It could be difficult to maintain. The new function parameters also need to be documented.